PythonVisualizations/SimpleSorting.py

```diff
@@ -205,34 +202,34 @@ def insert(self, val):
         self.moveItemsTo(cellPair, toPositions, steps=self.CELL_SIZE, sleepTime=0.01)
 
         # add a new Drawable with the new value, color, and display objects
-        self.list.append(drawable(
-            val, self.canvas.itemconfigure(cellPair[0], 'fill'), *cellPair))
+        self.list.append(drawable(val, self.canvas.itemconfigure(cellPair[0], 'fill'), *cellPair))
         callEnviron ^= set(cellPair) # Remove new cell from temp call environ
 
         # advance index for next insert
         self.highlightCodeTags('nitem_increment', callEnviron)
-        self.moveItemsBy(indexDisplay, (self.CELL_SIZE, 0), sleepTime=0.01)
+        self.moveItemsBy(self.nItems, (self.CELL_SIZE, 0), sleepTime=0.01)
 
         self.highlightCodeTags([], callEnviron)
         self.cleanUp(callEnviron)
 
     def removeFromEnd(self):
+        callEnviron = self.createCallEnvironment()        
+        
+        # pop a Drawable from the list
         if len(self.list) == 0:
             self.setMessage('Array is empty!')
             return

```

Comment:
Move the creation of the call environment to after the test for the empty array.  We want to avoid creating call stack environments that don't have a matching `self.cleanUp(callEnviron)` call.